docs/github-api.md

```diff
@@ -26,6 +30,54 @@ module.exports = robot => {
 
 See the [full API docs](https://octokit.github.io/rest.js/) to see all the ways you can interact with GitHub. Some API endpoints are not available on GitHub Apps yet, so check [which ones are available](https://developer.github.com/v3/apps/available-endpoints/) first.
 
+## GraphQL API
+
+> **Heads Up!** GraphQL support in Probot is currently a preview feature and this interface could change in a future releases without notice. [Let us know]() if you have any feedback or ideas to improve GraphQL support.
+
+The [GitHub GraphQL API](https://developer.github.com/v4/) is an alternative
+
+
+Here is an example of the same autoresponder app from above that comments on opened issues, but this time with GraphQL:
+
+```js
+// GraphQL query to get Node id for any resource, which is needed for mutations
+const getResource = `
+  query getResource($url: URI!) {
+    resource(url: $url) {
+      ... on Node {
+        id
+      }
+    }
+  }
+`
+
+// GraphQL query to add a comment
+const addComment = `
+  mutation comment($id: ID!, $body: String!) {
+    addComment(input: {subjectId: $id, body: $body}) {
+      clientMutationId
+    }
+  }
+`
+
+module.exports = robot => {
+  robot.on('issues.opened', async context => {
+    // Get the node id of the issue
+    const { resource } = await context.github.query(getResource, {
+      url: context.payload.issue.html_url
+    })
```

Comment:
So to do pretty much anything useful with GraphQL, you need to fetch the node id. this query works to look it up, but it fails pretty often with a `not found` error due to replication lag.

I know there's discussion of node id being included in webhooks by default so I'll pull on that thread and see what happens, but in the mean time…

 Does anyone have any ideas or know any tricks for getting the node id from a webhook payload?

Reply:
🤷‍♂️ This is the only way I've seen, since the node IDs are completely separate from any of the data coming over in a webhook.

Reply:
Yeah I think this is the only way to do this.

>but it fails pretty often with a `not found` error due to replication lag

This is interesting to me. I wonder if the graphql team knows about this and has any suggestions.